src/lamp_py/ad_hoc/runner_001.py

```diff
@@ -1,9 +1,126 @@
+import os
+from datetime import date
+from datetime import timedelta
+
+import pyarrow.parquet as pq
+
 from lamp_py.runtime_utils.process_logger import ProcessLogger
+from lamp_py.postgres.postgres_utils import start_rds_writer_process
+from lamp_py.runtime_utils.remote_files import S3_ARCHIVE
+from lamp_py.runtime_utils.remote_files import LAMP
+from lamp_py.runtime_utils.remote_files import S3_SPRINGBOARD
+from lamp_py.aws.s3 import file_list_from_s3
+from lamp_py.aws.s3 import download_file
+from lamp_py.ingestion.convert_gtfs_rt import GtfsRtConverter
+from lamp_py.ingestion.converter import ConfigType
+
+
+# pylint: disable=R0801
+class AdHocConverter(GtfsRtConverter):
+    """Custom ad-hoc Converter Class"""
+
+    def convert(self) -> None:
+        """
+        ad-hoc Convert process to add revenue field to parquet files
+
+        will read from Archive bucket and not move any files after processing
+        """
+        process_logger = ProcessLogger(
+            "ad_hoc_converter",
+            config_type=str(self.config_type),
+            file_count=len(self.files),
+        )
+        process_logger.log_start()
+
+        table_count = 0
+        try:
+            for table in self.process_files():
+                if table.num_rows == 0:
+                    continue
+
+                self.continuous_pq_update(table)
+                table_count += 1
+                process_logger.add_metadata(table_count=table_count)
+
+        except Exception as exception:
+            process_logger.log_failure(exception)
+        else:
+            process_logger.log_complete()
+        finally:
+            self.clean_local_folders()
+
+    def sync_with_s3(self, local_path: str) -> bool:
```

Comment:
question: Just curious, is this function used anywhere?

Reply:
Yes, it gets called in the base class: https://github.com/mbta/lamp/blob/42bb38b8afb459bf019542d54ffe5df04e98860f/src/lamp_py/ingestion/convert_gtfs_rt.py#L399

The `AdHocConverter` class just modifies the minimum functionality of the base class that allows these migration operations. 

The main modifications are removing `self.move_s3_files()` from the `convert` method. This would usually move files from our `incoming` S3 bucket to `archive`, but this process just reads straight from the `archive` bucket and moves nothing. 

And the `sync_with_s3` modifications that I put in the PR description.